packages/venia-ui/lib/components/FilterModal/FilterFooter/filterFooter.js

```diff
@@ -1,81 +1,45 @@
 import React, { useCallback } from 'react';
-import { object, shape, string } from 'prop-types';
-import { withRouter } from 'react-router-dom';
+import { shape, string } from 'prop-types';
+
 import { mergeClasses } from '../../../classify';
+import Button from '../../Button';
 import defaultClasses from './filterFooter.css';
-import { compose } from 'redux';
-import isObjectEmpty from '../../../util/isObjectEmpty';
-import { preserveQueryParams } from '@magento/peregrine/lib/util/preserveQueryParams';
-import { useCatalogContext } from '@magento/peregrine/lib/context/catalog';
-import { useAppContext } from '@magento/peregrine/lib/context/app';
-import serializeToParam from '@magento/peregrine/lib/util/serializeToParam';
 
 const FilterFooter = props => {
-    const { history, location } = props;
-    const [, { closeDrawer }] = useAppContext();
-    const [{ chosenFilterOptions }, catalogApi] = useCatalogContext();
-    const { clear: filterClear } = catalogApi.actions.filterOption;
-
-    const resetFilterOptions = useCallback(() => {
-        const queryParams = preserveQueryParams(location);
-        queryParams
-            ? history.push('?' + queryParams.toString())
-            : history.push();
-        filterClear();
-    }, [filterClear, history, location]);
-
-    const handleApplyFilters = useCallback(() => {
-        const queryParams = preserveQueryParams(location);
-        history.push(
-            '?' +
-                queryParams.toString() +
-                '&' +
-                serializeToParam(chosenFilterOptions)
-        );
-        closeDrawer();
-    }, [chosenFilterOptions, closeDrawer, history, location]);
-
     const classes = mergeClasses(defaultClasses, props.classes);
-    const areOptionsPristine = isObjectEmpty(chosenFilterOptions);
+    const { applyFilters, filterApi, filterState } = props;
+    const { clear } = filterApi;
+    const hasFilters = !!filterState.size;
 
-    const resetButtonClass = areOptionsPristine
-        ? classes.resetButtonDisabled
-        : classes.resetButton;
+    const handleApplyClick = useCallback(() => {
+        applyFilters();
+    }, [applyFilters]);
 
-    const applyButtonClass = areOptionsPristine
-        ? classes.applyButtonDisabled
-        : classes.applyButton;
+    const handleResetClick = useCallback(() => {
+        clear();
+        applyFilters();
```

Comment:
Yes, I meant to bring this up. This happens because `history.push()` and `closeDrawer()` are part of the same callback/effect now. We have to persist to the URL after resetting the filters or else the reset will be lost if they click the mask, and since I'm invoking the callback/effect to persist to the URL, it's also closing the drawer.

I'll see what happens if I rework the callback/effect.

Reply:
This has been fixed.